src/work-allocation-2/containers/work-case-list/work-case-list.component.spec.ts

```diff
@@ -0,0 +1,459 @@
+    fixture.detectChanges();
+
+    // check the case name is being sorted via ascending
+    expect(component.sortEvent.emit).toHaveBeenCalled();
+    expect(component.sortEvent.emit).toHaveBeenCalledWith('caseName');
+
+    // check that the case name is being sorted via descending
+    caseName.dispatchEvent(new Event('click'));
+    fixture.detectChanges();
+    expect(component.sortEvent.emit).toHaveBeenCalled();
+    expect(component.sortEvent.emit).toHaveBeenCalledWith('caseName');
+
+    // click the second example button and verify that sorting is for case category
+    categoryButton.dispatchEvent(new Event('click'));
+    fixture.detectChanges();
+    expect(component.sortEvent.emit).toHaveBeenCalled();
+    expect(component.sortEvent.emit).toHaveBeenCalledWith('caseCategory');
+
+    // click the first example button and verify that sorting is again for case reference
+    startDateButton.dispatchEvent(new Event('click'));
+    fixture.detectChanges();
+    expect(component.sortEvent.emit).toHaveBeenCalled();
+    expect(component.sortEvent.emit).toHaveBeenCalledWith('startDate');
+  });
+
+  describe('act upon deep linking', () => {
+    const id = '12345678';
+
+    it('should select appropriate case from location hash', () => {
+      spyOnProperty(mockRouter, 'url', 'get').and.returnValue(`caseList#manage_${id}`);
+      const navigateCallsBefore = mockRouter.navigateCalls.length;
+      const caseItem = { id } as Case;
+      wrapper.cases = [ caseItem ];
+      fixture.detectChanges();
+      expect(component.getSelectedCase()).toEqual(caseItem);
+      expect(mockRouter.navigateCalls.length).toBeGreaterThan(navigateCallsBefore);
+      const lastNavigateCall = mockRouter.navigateCalls.pop();
+      expect(lastNavigateCall).toBeDefined();
+      expect(lastNavigateCall.commands).toEqual([ 'caseList' ]);
+      expect(lastNavigateCall.extras).toEqual({ fragment: `manage_${id}` });
+    });
+
+    it('should handle a location hash for a case that does not exist', () => {
+      spyOnProperty(mockRouter, 'url', 'get').and.returnValue(`caseList#manage_${id}`);
+      const navigateCallsBefore = mockRouter.navigateCalls.length;
+      const caseItem = { id: '99999999' } as Case;
+      wrapper.cases = [ caseItem ];
+      fixture.detectChanges();
+      expect(component.getSelectedCase()).toBeNull();
+      expect(mockRouter.navigateCalls.length).toBeGreaterThan(navigateCallsBefore);
+      const lastNavigateCall = mockRouter.navigateCalls.pop();
+      expect(lastNavigateCall).toBeDefined();
+      expect(lastNavigateCall.commands).toEqual([ 'caseList' ]);
+      expect(lastNavigateCall.extras).toBeUndefined();
+    });
+  });
+
+  describe('generate pagination summary', () => {
+    let paginationSummary: HTMLElement;
+
+    beforeEach(() => {
+      paginationSummary = fixture.debugElement.nativeElement.querySelector('#search-result-summary__text');
+    });
+
+    it('should correctly set the summary text', () => {
```

Comment:
_Question:_
Do you intend to have more tests in here? If not, the `beforeEach(...)` is kinda pointless.

Reply:
Potentially...